### PR TITLE
Fixing Inventories, and Trade Offers.

### PR DESCRIPTION
**Pull request by bartico6**
Fixing the following:
1. Empty/null inventories.
2. Trade Offers generated without all items (omitting "missing" assets for no reason)
The missing assets should be added to the trade offers for user's verification for instance when an offer goes "Invalid", a user might want to know which items went missing to re-offer, right? Well the missing assets were omitted and lost while converting a JSON response to a TradeOffer class.
3. Added a nullcheck for the response in TradeOfferManager - so SteamBot doesn't crash after receiving an empty response which - let's be honest - happens a lot with Steam.

### CODE REVIEW

File: SteamTrade/TradeOffer/TradeOfferManager.cs [hunk 1]
```diff
@@ -167,10 +167,15 @@ public bool GetOffer(string offerId, out TradeOffer tradeOffer)
                 else
                 {
                     //todo: log steam api is giving us invalid offers.
-                    Debug.WriteLine("Offer returned from steam api is not valid : " + resp.Offer.TradeOfferId);
+                    Debug.WriteLine("Received invalid offer from Steam API");
+//                    Logging.Log("resp.Offer stringified: ->" + (resp.Offer == null ? "Null" : resp.Offer.ToString()), Logging.LogLevel.ERROR);
```
Comment (scholtzm) [inline SteamTrade/TradeOffer/TradeOfferManager.cs]: This line should be removed.

Comment (bartico6) [inline SteamTrade/TradeOffer/TradeOfferManager.cs]: Done!

File: SteamTrade/Inventory.cs [hunk 2]
```diff
@@ -15,11 +15,26 @@ public class Inventory
...
+                {
+                    result = JsonConvert.DeserializeObject<InventoryResponse>(response);
+                }
+                catch (Exception e)
+                {
+
+                }
+                attempts++;
+            }
+            if (result == null || result.result == null || result.result.items == null)
```
Comment (scholtzm) [inline SteamTrade/Inventory.cs]: This should just throw an exception at this point. Otherwise this is a breaking change.

Comment (bartico6) [inline SteamTrade/Inventory.cs]: Removed the try-catch and added an exception to throw if the inventory cannot be fetched.

**Comment by bartico6** [comment]
Oh cool appveyor decided to error out on
> Installing 'NUnit 2.6.4'.
Successfully installed 'NUnit 2.6.4'.
Command exited with code 1

And say it failed! Aw yis!

File: SteamTrade/TradeOffer/TradeOffer.cs [hunk 3]
```diff
@@ -119,7 +105,7 @@ public bool CounterOffer(out string newTradeOfferId, string message = "")
         /// <param name="offerId">The trade offer id if successully created</param>
         /// <param name="message">Optional message to included with the trade offer</param>
         /// <returns>true if successfully sent, otherwise false</returns>
-        public bool Send(out string offerId, string message = "")
+        public int Send(out string offerId, string message = "")
```
Comment (scholtzm) [inline SteamTrade/TradeOffer/TradeOffer.cs]: Why this breaking change?

File: SteamTrade/TradeOffer/TradeOffer.cs [hunk 4]
```diff
@@ -138,16 +124,16 @@ public bool Send(out string offerId, string message = "")
         /// <param name="token">The token of the partner</param>
         /// <param name="message">Optional message to included with the trade offer</param>
         /// <returns></returns>
-        public bool SendWithToken(out string offerId, string token, string message = "")
+        public int SendWithToken(out string offerId, string token, string message = "", bool debug = false)
```
Comment (scholtzm) [inline SteamTrade/TradeOffer/TradeOffer.cs]: Another change here?

File: SteamTrade/TradeOffer/TradeOffer.cs [hunk 5]
```diff
@@ -138,16 +124,16 @@ public bool Send(out string offerId, string message = "")
...
         {
             offerId = String.Empty;
             if (TradeOfferId == null)
             {
-                return Session.SendTradeOfferWithToken(message, PartnerSteamId, this.Items, token, out offerId);
+                return Session.SendTradeOfferWithToken(message, PartnerSteamId, this.Items, token, out offerId, debug);
```
Comment (scholtzm) [inline SteamTrade/TradeOffer/TradeOffer.cs]: Does this method even exist?

https://github.com/Jessecar96/SteamBot/blob/master/SteamTrade/TradeOffer/OfferSession.cs#L221

File: SteamTrade/TradeOffer/TradeOffer.cs (hunk 4 above)
Comment (BlueRaja) [inline SteamTrade/TradeOffer/TradeOffer.cs]: What's with all the unused `debug` parameters?

**Comment by bartico6** [comment]
Genuinely sorry for the fails in this PR. :cry:

**Comment by scholtzm** [comment]
Why is AppVeyor failing?

Also, the commits in this PR should be squashed before merging imo.

**Comment by bartico6** [comment]
Squashed = compressed to one commit or waht?

**Comment by scholtzm** [comment]
Yeah, squashed into a single commit.

**Comment by bartico6** [comment]
How do I do so? (I'm fairly new to git)

**Comment by BlueRaja** [comment]
http://blogs.atlassian.com/2014/06/interactive-rebase-sourcetree/
http://stackoverflow.com/questions/5509543/how-do-i-properly-force-a-git-push

**Comment by scholtzm** [comment]
I found this guide to be very straight-forward: https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request

**Comment by bartico6** [comment]
Since apparently my fork is behind the official repo a few commits, I will close this PR, and manually open another one once all of the "stuff" is done. Sorry.

**Pull request closed without merging**